cico_run_tests.sh

```diff
@@ -21,6 +21,13 @@ docker run --detach=true -t \
 docker exec -t "$BUILDER-run" bash -ec 'go get github.com/jteeuwen/go-bindata/...'
 docker exec -t "$BUILDER-run" bash -ec 'go generate'
 docker exec -t "$BUILDER-run" bash -ec 'go build -o dist/traefik ./cmd/traefik'
-docker exec -t "$BUILDER-run" bash -ec 'go test -v ./middlewares/osio/'
-docker exec -t "$BUILDER-run" bash -ec 'go test -v ./provider/osio/'
+
+docker exec -t "$BUILDER-run" bash -ec 'go test -v ./middlewares/osio/ -coverprofile coverage.middlewares -covermode=set -timeout 5m'
+docker exec -t "$BUILDER-run" bash -ec 'go test -v ./provider/osio/ -coverprofile coverage.provider -covermode=set -timeout 5m'
```

Comment:
@aslakknutsen @alexeykazakov IIRC we had quite some difficulties to get a correct cross-package coverage. So if a function in `provide/osio` tests part of `middlewares/osio`, then this bit of coverage was not detected. I wonder if codecov now does a good job with this. We had to merge coverage files using https://github.com/wadey/gocovmerge in order to get proper coverage.

@nurali-techie can you do an experiment and write a function inside of the one package that you test in the other and see if the coverage is correctly calculated? Then maybe we can clean our coverage calculation a bit as well.

Reply:
@kwk I already have stuff where pkg `middlewares/osio` uses `provide/osio`.

When I run below
`go test -v ./middlewares/osio/ -coverprofile coverage.middlewares -covermode=set`
It won't include coverage for pkg `provide/osio`.

But when I run below with `coverpkg`
`go test -v ./middlewares/osio/ -coverprofile coverage.middlewares -covermode=set -coverpkg github.com/containous/traefik/middlewares/osio,github.com/containous/traefik/provider/osio`
It include coverage for pkg `provide/osio`

And above is all golang stuff and it has nothing to do with codecov.